src/axios-har-tracker.ts

```diff
@@ -1,112 +1,204 @@
+        cookies: this.getCookies(JSON.stringify(response.config.headers['Cookie'])),
+        bodySize: JSON.stringify(response.data).length,
+        redirectURL: '',
+        headersSize: -1,
+        content: {
+          size: JSON.stringify(response.data).length,
+          mimeType: response.headers['content-type'] ? response.headers['content-type'] : 'text/plain',
+          text: JSON.stringify(response.data)
+        },
+        cache: {},
+        timings: {
+          blocked: -1,
+          dns: -1,
+          ssl: -1,
+          connect: -1,
+          send: 10,
+          wait: 10,
+          receive: 10,
+          _blocked_queueing: -1
+        }
+      }
+      return responseObject;
+    } else {
+      let responseObject = {
+        status: [],
+        statusText: [],
         headers: [],
-        queryString: getParams(config.params),
+        startedDateTime: [],
+        time: [],
+        httpVersion: [],
+        cookies: [],
+        bodySize: 0,
+        redirectURL: '',
         headersSize: -1,
-        bodySize: -1
-      };
-      return config;
-    },
-    error => {
-      return Promise.reject(error);
+        content: {
+          size: 0,
+          mimeType: 'text/plain',
+          text: ''
+        },
+        cache: {},
+        timings: {
+          blocked: -1,
+          dns: -1,
+          ssl: -1,
+          connect: -1,
+          send: 10,
+          wait: 10,
+          receive: 10,
+          _blocked_queueing: -1
+        }
+      }
+      return responseObject;
     }
-  );
-
-  axios.interceptors.response.use(
-    async resp => {
-      if (resp) {
-        newEntry.response = {
-          status: resp.status,
-          statusText: resp.statusText,
-          headers: getHeaders(resp.headers),
-          startedDateTime: new Date(resp.headers.date),
-          time: resp.headers['request-duration'] = Math.round(
-            process.hrtime(resp.headers['request-startTime'])[0] * 1000 +
-              process.hrtime(resp.headers['request-startTime'])[1] / 1000000
-          ),
-          httpVersion: `HTTP/${resp.request.res.httpVersion}`,
-          cookies: getCookies(JSON.stringify(resp.config.headers['Cookie'])),
-          bodySize: JSON.stringify(resp.data).length,
-          redirectURL: '',
-          headersSize: -1,
-          content: {
-            size: JSON.stringify(resp.data).length,
-            mimeType: resp.headers['content-type'] ? resp.headers['content-type'] : 'text/plain',
-            text: JSON.stringify(resp.data)
-          },
-          cache: {},
-          timings: {
-            blocked: -1,
-            dns: -1,
-            ssl: -1,
-            connect: -1,
-            send: 10,
-            wait: 10,
-            receive: 10,
-            _blocked_queueing: -1
-          }
-        };
-        const enteriesContent = Object.assign({}, newEntry);
-        generatedHar.log.entries.push(enteriesContent);
+  }
+
+  private pushNewEntryResponse(response) {
```

Comment:
Why a separate method for this? It is used only once

Reply:
can be used also under response